File: algorithms/python/productOfArrayExceptSelf.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        answer = [1]*len(nums)
        prefix = 1
        for i in range(len(nums)):
            answer[i] = prefix
            prefix *= nums[i]
            
        
        postfix = 1
        for i in range(len(nums) - 1,-1, -1):
            answer[i] *= postfix
            postfix *= nums[i]
            
        return answer

File: algorithms/java/productOfArrayExceptSelf.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int p = 1;
        int[] res = new int[nums.length];
        for(int i=0;i<nums.length; i++) {
            res[i] = p;
            p *= nums[i];
        }
        p = 1;
        for(int i=nums.length-1; i >= 0; i--) {
            res[i] *= p;  
            p *= nums[i];
        }

        return res;
    }
}